packages/browser/src/browser/index.ts

```diff
@@ -353,7 +353,14 @@ async function loadAnalytics(
   const plugins = settings.plugins ?? []
 
   const classicIntegrations = settings.classicIntegrations ?? []
-  Stats.initRemoteMetrics(legacySettings.metrics)
+
+  if (options.metricsEndpoint && legacySettings.metrics) {
+    legacySettings.metrics.host = options.metricsEndpoint
+  }
+
+  Stats.initRemoteMetrics(
+    legacySettings.metrics ?? { host: options.metricsEndpoint }
+  )
```

Comment:
What happens if `Stats.initRemoteMetrics` receives `{ host: undefined }` ? Given that we were always sending the `legacySettings.metrics` object which looks like could be either full or undefined.

I'm also curious if we need this coalescing here as just above we're already overriding it.